src/mlagility/api/model_api.py

```diff
@@ -116,10 +123,71 @@ def benchmark_model(
                 )
                 perf = cpu_model.benchmark(backend=backend)
 
+        elif device == "x86" and runtime in ["torch", "torch_compiled"]:
+
+            # Create cache folder with stats file
+            # Although building with an empty sequence is possible, we don't want
+            # the model to be rebuilt when other devices are used, causing the
+            # results inside of the stats file to be lost.
+            build_model(
+                model=model,
+                inputs=inputs,
+                build_name=build_name,
+                cache_dir=cache_dir,
+                rebuild=rebuild,
+                sequence=sequence,
+            )
+
+            torch_version = torch.__version__
+            if runtime == "torch_compiled":
+                clean_torch_version = torch_version.split("+")[0]
+                if version.parse(clean_torch_version) < version.parse("2.0.0"):
+                    BenchmarkException(
+                        (
+                            f"{runtime} can only be used with Pytorch 2.0.0 or above. "
+                            f"However, version {torch_version} was found."
+                        )
+                    )
+
+                model = torch.compile(model)
+
+            if not build_only:
+                repetitions = 100
+                total_time = [0] * repetitions
+                for idx in range(repetitions):
+                    start_time = time.process_time()
+                    model(**inputs)
+                    end_time = time.process_time()
+                    total_time[idx] = end_time - start_time
+
+                # Calculate perf from total_time
+                mean_latency_ms = mean(total_time) * 1000
+                throughput_ips = float(1 / (np.sum(total_time) / repetitions))
+
+                if runtime == "torch":
+                    device_name = (
+                        get_cpu_specs()["CPU Name"] + f" (Pytorch {torch_version})"
+                    )
+                else:
+                    device_name = (
+                        get_cpu_specs()["CPU Name"]
+                        + f" (Pytorch {torch_version} Compiled)"
+                    )
+
+                return MeasuredPerformance(
+                    mean_latency=mean_latency_ms,
+                    throughput=throughput_ips,
+                    device=device_name,
+                    device_type=device,
+                    build_name=build_name,
+                )
```

Comment:
Suggest adding a `runtime` field to `MeasuredPerformance` instead of concatenating the `device` and `runtime` values together into the `device` field. That will make it easier to query the `runtime` by itself later, if needed.

Reply:
Things are a lot more organized now! 

```
performance:
  Intel(R) Xeon(R) CPU @ 2.20GHz (ort v1.14.1):
    build_name: linear_selftest_d5b1df11
    device: Intel(R) Xeon(R) CPU @ 2.20GHz
    device_type: x86
    latency_units: milliseconds (ms)
    mean_latency: 0.016006752848625183
    runtime: ort
    runtime_version: 1.14.1
    throughput: 62473.632813409116
    throughput_units: inferences per second (IPS)
  Intel(R) Xeon(R) CPU @ 2.20GHz (torch v2.0.0+cu117):
    build_name: linear_selftest_d5b1df11
    device: Intel(R) Xeon(R) CPU @ 2.20GHz
    device_type: x86
    latency_units: milliseconds (ms)
    mean_latency: 0.02581126056611538
    runtime: torch
    runtime_version: 2.0.0+cu117
    throughput: 38742.78040154243
    throughput_units: inferences per second (IPS)
```